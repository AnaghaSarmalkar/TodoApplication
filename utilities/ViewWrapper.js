import React from "react";
import { View } from "react-native";

export default function ViewWrapper(props) {
  return (
    <View
      {...props}
      // style={{
      //   flex: 1,
      //   alignItems: "center",
      //   justifyContent: "center",
      //   // paddingTop: insets.top,
      // }}
    >
      {props.children}
    </View>
  );
}
