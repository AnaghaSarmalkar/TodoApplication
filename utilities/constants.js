import { Platform } from "react-native";

export const isIos = Platform.OS === "ios";
export const isWeb = Platform.OS === "web";
export const isAndroid = Platform.OS === "android";
