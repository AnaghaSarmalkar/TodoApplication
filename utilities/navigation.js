export function setPageTitle(navigation, title) {
  if (title) {
    navigation.setOptions({
      title: title,
    });
  }
}
