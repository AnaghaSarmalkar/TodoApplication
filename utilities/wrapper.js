import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";

function SafeAreaWrapper({ children }) {
  <SafeAreaView>{children}</SafeAreaView>;
}
