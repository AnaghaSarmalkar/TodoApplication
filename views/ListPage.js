import React from "react";
import List from "../components/List/List";

export default function ListPage({ data }) {
  return <List data={data} />;
}
