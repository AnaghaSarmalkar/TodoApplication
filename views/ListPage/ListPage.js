import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";
import { Header } from "react-native/Libraries/NewAppScreen";
import Error from "../../components/Error/Error";
import Item from "../../components/Item/Item";
import List from "../../components/List/List";
import Loading from "../../components/Loading/Loading";

export default function ListPage({ data, loading, error }) {
  if (loading) return <Loading />;
  if (error) return <Error message={error.message} />;
  const header = <Header title="Todo List" />;

  const renderItem = ({ item }) => <Item item={item} />;
  return (
    <SafeAreaView>
      <List
        data={data.locations}
        headerComponent={header}
        renderItem={renderItem}
      />
    </SafeAreaView>
  );
}
