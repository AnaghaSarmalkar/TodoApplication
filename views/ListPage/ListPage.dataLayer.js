import { gql, useQuery } from "@apollo/client";
import React from "react";
import ListPage from "./ListPage";

export const getLocationsSimple = gql`
  query GetLocationsSimple {
    locations {
      id
      name
    }
  }
`;

export default function ListPageDataLayer() {
  const { loading, error, data } = useQuery(getLocationsSimple);

  return <ListPage data={data} loading={loading} error={error} />;
}
