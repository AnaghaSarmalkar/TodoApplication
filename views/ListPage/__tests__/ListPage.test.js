import React from "react";
import { shallow } from "../../../utilities/testing";
import ListPage from "../ListPage";

describe("ListPage", () => {
  it("renders loading component", async () => {
    const wrapper = shallow(
      <ListPage data={undefined} loading={true} error={undefined} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders error component", async () => {
    const error = new Error("An error occured.");
    const wrapper = shallow(
      <ListPage data={undefined} loading={false} error={error} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders List component", async () => {
    const data = {
      data: {
        locations: [
          { id: "1", name: "Test Location 1" },
          { id: "2", name: "Test Location 2" },
          { id: "3", name: "Test Location 3" },
        ],
      },
    };

    const wrapper = shallow(
      <ListPage data={data} loading={false} error={undefined} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
