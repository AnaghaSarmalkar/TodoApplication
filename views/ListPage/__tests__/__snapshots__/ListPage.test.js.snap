// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage renders List component 1`] = `<List />`;

exports[`ListPage renders error component 1`] = `
<Error
  message="An error occured."
/>
`;

exports[`ListPage renders loading component 1`] = `<Loading />`;
